compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -918,7 +917,40 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                         gen_args.span(),
                         "parenthesized generic arguments cannot be used in associated type constraints"
                     );
-                    // FIXME: try to write a suggestion here
+                    if let Ok(snippet) = self.sess.source_map().span_to_snippet(data.span) {
+                        // Suggest replacing parentheses with angle brackets `Trait(params...)` to `Trait<params...>`
+                        if !data.inputs.is_empty() {
```

Comment:
swap these two if/else blocks so that the condition does not need to get inverted.